src/appsettings/settings.py

```diff
@@ -807,25 +887,38 @@ def __init__(self, name='', default=dict, required=False,
                 the setting's prefix (overrides ``AppSettings.Meta`` prefix).
             call_default (bool): whether to call the default (if callable).
             transform_default (bool): whether to transform the default value.
-            **checker_kwargs: keyword arguments passed to the checker.
-        """
+            validators (list of callables): list of additional validators to use.
+            key_type: Noop. Deprecated.
+            value_type (type): the type of the items inside the iterable.
+            min_length (int): minimum length of the iterable (included).
+            max_length (int): maximum length of the iterable (included).
+            empty (bool): whether empty iterable is allowed. Deprecated in favor of min_length.
+        """
+        if key_type is not None:
+            warnings.warn("Argument key_type does nothing and is deprecated.", DeprecationWarning)
```

Comment:
Why should it be deprecated?

Reply:
I haven't found any use case nor restriction to maintain it. Objects which can serve as a key in dictionary are limited to hashables by python and I don't actually see any benefits in further restriction.

I'm not very strong about this, so if you have any arguments to keep it, I'm open to reconsider.

Reply:
OK, The integers make a good point.